mlir/utils/jenkins/Jenkinsfile.downstream

```diff
@@ -41,8 +46,27 @@ pipeline {
                         -DMLIR_ROCM_RUNNER_ENABLED=1 \
                         -DMLIR_MIOPEN_DRIVER_ENABLED=1
 
-                    # build LLVM / MLR and run tests
-                    cmake --build . --target check-mlir
+	     	    # premerge checks
+                    cd ..
+                    if [ ! -f ./compile_commands.json ]; then
+                       ln -s build/compile_commands.json compile_commands.json
+                    fi
+		    python3.7 ./mlir/utils/jenkins/static-checks/premerge-checks.py
```

Comment:
Why does the CI show the following:
```
+ python3.7 ./mlir/utils/jenkins/static-checks/premerge-checks.py
no modified files to format
```
Is it because you are changing the jenkins folder, which clang static checks doesn't care about? If that's the case, we might need a case where manually introduce a change that can make clang static checks to break. Just to make sure the static check can actually work.

Reply:
It would be better if we can introduce static checks to go through the files under jenkins folder.

Reply:
By default, clang-format formats C/C++/Java/JavaScript/Objective-C/Protobuf/C# code and skips other types of files. I tried to ran clang-format on premerge-checks.py, it generates weird and incorrect results. For example, it added spaces between words in #!/usr/bin/env phthon3. Looks like it treated this line as a normal comment. And it didn't fix inconsistent indents. I think applying static checks to jenkins folder needs
 better understanding of how clang-format and clang-tidy work and should belong to another review.

Reply:
I have manually changed a mlir source file and verified that clang-format and clang-tidy would stop and issue an exit code of 1, as expected.

Reply:
Thanks for helping to figure out the file type part. That sounds reasonable to skip the python file check as upstream do.

Regarding on manually introducing the change, if you don't think that's necessary, then could you keep an closer eye on the CI state after your submission? Just to make sure everything works as expected.

Reply:
Sure. I will.